smarts/core/sumo_traffic_simulation.py

```diff
@@ -118,7 +120,10 @@ def __init__(
         self._current_reload_count = 0
         # /TODO
 
-        self._traci_exceptions = (TraCIException, FatalTraCIError)
+        self._traci_exceptions = (
+            traci.exceptions.TraCIException,
+            traci.exceptions.FatalTraCIError,
```

Comment:
Yeah, this change was forced upon me because the existing code (including in the `develop` branch) fails to catch Traci exceptions.  

(For example, I was getting an exception raised in `teardown()` due to the `self._remove_vehicles()` there, and the `except` clause there wasn't catching it.  Exceptions in `_initialize_traci_conn()` were also not being caught correctly.  I found through experimentation that it was required to be explicit with the exception names in order to be caught, so I changed it.  I didn't spend too much time trying to figure out _why_ this was the case though!)

I'm fine with changing it to whatever you think is best, but of course it has to actually _work_ (i.e., the exceptions have to be catchable!).